src/confd/src/confd/ietf-interfaces.c

```diff
@@ -541,24 +728,53 @@ static int netdag_gen_afspec_set(struct dagger *net, struct lyd_node *dif,
 
 	DEBUG_IFACE(dif, "");
 
+	if (!strcmp(iftype, "iana-if-type:bridge"))
+		return netdag_set_bridge(net, dif, cif, ip);
 	if (!strcmp(iftype, "iana-if-type:l2vlan"))
 		return netdag_gen_vlan(net, dif, cif, ip);
+	if (!strcmp(iftype, "infix-if-type:veth"))
+		return 0;
 
 	ERROR("Unable to configure unsupported interface type \"%s\"", iftype);
 	return -ENOSYS;
 }
 
+static int netdag_bridge_links(struct dagger *net, struct lyd_node *dif,
+			       struct lyd_node *cif, FILE *ip)
+{
+	const char *ifname = lydx_get_cattr(cif, "name");
+	struct lyd_node *node;
+	int err = 0;
+
+	node = lydx_get_descendant(lyd_child(cif), "bridge-port", NULL);
+	if (node) {
+		const char *brname = lydx_get_cattr(node, "bridge");
+
+		fprintf(ip, " master %s", brname);
+
+		err = dagger_add_dep(net, ifname, brname);
```

Comment:
Isn't this still inverted? The bridge should depend on all its ports being setup before running its own setup?

I think the `link set dev PORT master BR` needs to go in an `.ip` file in the _bridge's_ action directory (at a priority that runs after the bridge itself is created, but before applying VLANs etc.

Also, the fact that `ietf-interfaces.c` does not contains the string `nomaster` has me worried that we're not handling the case when interfaces are removed from a bridge? Correspondingly, that command should also run in the _bridge's_ exit action, I think.